apps/src/templates/SchoolNotFound.jsx

```diff
@@ -0,0 +1,201 @@
+import React, { Component, PropTypes } from 'react';
+import color from "../util/color";
+import i18n from "@cdo/locale";
+import { STATES } from '../geographyConstants';
+
+const styles = {
+  question: {
+    fontSize: 16,
+    fontFamily: '"Gotham 3r", sans-serif',
+    color: color.charcoal,
+    paddingTop: 10,
+    paddingBottom: 5
+  },
+  input: {
+    height: 40,
+    width: 250,
+    fontFamily: '"Gotham 3r", sans-serif',
+    padding: 5
+  },
+  dropdown: {
+    fontFamily: '"Gotham 4r", sans-serif',
+    color: color.charcoal,
+    height: 30,
+    marginTop: 5,
+    width: 250
+  },
+  errors: {
+    fontSize: 14,
+    fontFamily: '"Gotham 3r", sans-serif',
+    color: color.red,
+    paddingTop: 5,
+    paddingBottom: 5
+  },
+  field: {
+    float: 'left',
+    height: '80px',
+    width: '450px',
+  },
+  clear: {
+    width: '100%',
+    clear: 'both'
+  },
+  asterisk: {
+    fontSize: 20,
+    fontFamily: '"Gotham 5r", sans-serif',
+    color: color.red,
+  },
+};
+
+export default class SchoolNotFound extends Component {
+  static propTypes = {
+    setField: PropTypes.func,
+    schoolName: PropTypes.string,
+    schoolCity: PropTypes.string,
+    schoolState: PropTypes.string,
+    schoolZip: PropTypes.string,
+    schoolType: PropTypes.string,
+    showErrorMsg: PropTypes.bool
+  };
+
+  sendNameToParent = (event) => {
+    this.props.setField("schoolName", event.target.value);
+  }
+
+  sendTypeToParent = (event) => {
+    this.props.setField("schoolType", event.target.value);
+  }
+
+  sendStateToParent = (event) => {
+    this.props.setField("schoolState", event.target.value);
+  }
+
+  sendZipToParent = (event) => {
+    this.props.setField("schoolZip", event.target.value);
+  }
+
+  sendCityToParent = (event) => {
+    this.props.setField("schoolCity", event.target.value);
+  }
```

Comment:
These are pretty much the same. I should pass the field as a parameter and combine these functions into one.

Reply:
:+1: Yeah!  Basically one `onChange` function for the whole form.  Then your tests can pass a spy `onChange` and detect that it gets called correctly when events are triggered.